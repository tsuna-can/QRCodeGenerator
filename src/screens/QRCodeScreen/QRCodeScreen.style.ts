import {StyleSheet} from 'react-native';
import COLORS from '../../theme/colors';
import {FONT_SIZES} from '../../theme/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 10,
  },
  textContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    marginHorizontal: 20,
  },
  text: {
    fontSize: FONT_SIZES.xxxxLarge,
    color: COLORS.BLACK,
    textAlignVertical: 'center',
    textAlign: 'center',
  },
  qrCodeContainer: {
    flex: 5,
    margin: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
  buttonContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'flex-end',
    padding: 10,
  },
});

export default styles;
