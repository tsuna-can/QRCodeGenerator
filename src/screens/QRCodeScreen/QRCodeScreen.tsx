import React, {useState, useRef, useMemo} from 'react';
import {View, Text, SafeAreaView} from 'react-native';
import Button from '../../components/button/button';
import styles from './QRCodeScreen.style';
import QRCodeView from '../../components/QRCodeView/QRCodeView';
import type {NativeStackScreenProps} from '@react-navigation/native-stack';
import type {RootStackParamList} from '../../App';
import {zeroPaddiong, calcMaxValue, calcMinValue} from '../../utils/padding';
import Icon from 'react-native-vector-icons/FontAwesome';

type Props = NativeStackScreenProps<RootStackParamList, 'QRCode'>;

const QRCodeScreen = ({navigation, route}: Props) => {
  const fixedValue = useRef(route.params.fixedValue);
  const [variableValue, setVariableValue] = useState(
    route.params.variableValue,
  );
  const digits = useRef(route.params.digits);

  const minValue = useMemo(() => calcMinValue(digits.current), [digits]);
  const lastValue = useMemo(() => calcMaxValue(digits.current), [digits]);

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.textContainer}>
        <Text
          style={styles.text}>{`${fixedValue.current}${variableValue}`}</Text>
      </View>
      <View style={styles.qrCodeContainer}>
        <QRCodeView
          value={`${fixedValue.current}${variableValue}`}
          size={250}
        />
      </View>
      <View style={styles.buttonContainer}>
        <Button
          disabled={variableValue === minValue}
          onClick={() => {
            setVariableValue(curr => {
              const next = Math.max(Number(curr) - 1, 0);
              return zeroPaddiong(next.toString(), digits.current);
            });
          }}
          icon={<Icon name="arrow-left" size={50} color="white" />}
        />
        <Button
          disabled={variableValue === lastValue}
          onClick={() => {
            setVariableValue(curr => {
              const next = Math.min(Number(curr) + 1, Number(lastValue));
              return zeroPaddiong(next.toString(), digits.current);
            });
          }}
          icon={<Icon name="arrow-right" size={50} color="white" />}
        />
      </View>
    </SafeAreaView>
  );
};

export default QRCodeScreen;
