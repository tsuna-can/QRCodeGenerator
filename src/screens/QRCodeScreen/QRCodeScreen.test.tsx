import React from 'react';
import renderer from 'react-test-renderer';
import QRCodeScreen from './QRCodeScreen';

const navigation = {
  navigate: jest.fn(),
};

const route = {
  params: {
    fixedValue: 'test-fixed-value',
    variableValue: 'test-variable-value',
    digits: 1,
  },
};

describe('HomeScreen', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<QRCodeScreen navigation={navigation} route={route} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
