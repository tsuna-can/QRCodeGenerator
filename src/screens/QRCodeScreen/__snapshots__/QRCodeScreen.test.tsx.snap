// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "margin": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 40,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    >
      test-fixed-valuetest-variable-value
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 5,
        "justifyContent": "center",
        "margin": 10,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="250"
      bbWidth="250"
      focusable={false}
      height={250}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 250,
            "width": 250,
          },
        ]
      }
      vbHeight={250}
      vbWidth={250}
      width={250}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              [
                0,
                -65536,
                1,
                -16711681,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="grad"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="100%"
          />
        </RNSVGDefs>
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGRect
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            height="250"
            propList={
              [
                "fill",
              ]
            }
            width="250"
            x="0"
            y="0"
          />
        </RNSVGGroup>
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGPath
            d="M0 4.310344827586207 L60.344827586206904 4.310344827586207 M77.58620689655173 4.310344827586207 L86.20689655172414 4.310344827586207 M94.82758620689656 4.310344827586207 L103.44827586206898 4.310344827586207 M129.31034482758622 4.310344827586207 L146.55172413793105 4.310344827586207 M155.17241379310346 4.310344827586207 L181.0344827586207 4.310344827586207 M189.6551724137931 4.310344827586207 L250.00000000000003 4.310344827586207 M0 12.931034482758623 L8.620689655172415 12.931034482758623 M51.72413793103449 12.931034482758623 L60.344827586206904 12.931034482758623 M77.58620689655173 12.931034482758623 L86.20689655172414 12.931034482758623 M155.17241379310346 12.931034482758623 L163.79310344827587 12.931034482758623 M189.6551724137931 12.931034482758623 L198.27586206896552 12.931034482758623 M241.37931034482762 12.931034482758623 L250.00000000000003 12.931034482758623 M0 21.551724137931036 L8.620689655172415 21.551724137931036 M17.24137931034483 21.551724137931036 L43.10344827586207 21.551724137931036 M51.72413793103449 21.551724137931036 L60.344827586206904 21.551724137931036 M68.96551724137932 21.551724137931036 L94.82758620689656 21.551724137931036 M103.44827586206898 21.551724137931036 L112.0689655172414 21.551724137931036 M120.68965517241381 21.551724137931036 L129.31034482758622 21.551724137931036 M137.93103448275863 21.551724137931036 L146.55172413793105 21.551724137931036 M163.79310344827587 21.551724137931036 L181.0344827586207 21.551724137931036 M189.6551724137931 21.551724137931036 L198.27586206896552 21.551724137931036 M206.89655172413796 21.551724137931036 L232.7586206896552 21.551724137931036 M241.37931034482762 21.551724137931036 L250.00000000000003 21.551724137931036 M0 30.172413793103452 L8.620689655172415 30.172413793103452 M17.24137931034483 30.172413793103452 L43.10344827586207 30.172413793103452 M51.72413793103449 30.172413793103452 L60.344827586206904 30.172413793103452 M68.96551724137932 30.172413793103452 L77.58620689655173 30.172413793103452 M86.20689655172414 30.172413793103452 L94.82758620689656 30.172413793103452 M146.55172413793105 30.172413793103452 L155.17241379310346 30.172413793103452 M172.41379310344828 30.172413793103452 L181.0344827586207 30.172413793103452 M189.6551724137931 30.172413793103452 L198.27586206896552 30.172413793103452 M206.89655172413796 30.172413793103452 L232.7586206896552 30.172413793103452 M241.37931034482762 30.172413793103452 L250.00000000000003 30.172413793103452 M0 38.793103448275865 L8.620689655172415 38.793103448275865 M17.24137931034483 38.793103448275865 L43.10344827586207 38.793103448275865 M51.72413793103449 38.793103448275865 L60.344827586206904 38.793103448275865 M68.96551724137932 38.793103448275865 L77.58620689655173 38.793103448275865 M112.0689655172414 38.793103448275865 L137.93103448275863 38.793103448275865 M163.79310344827587 38.793103448275865 L181.0344827586207 38.793103448275865 M189.6551724137931 38.793103448275865 L198.27586206896552 38.793103448275865 M206.89655172413796 38.793103448275865 L232.7586206896552 38.793103448275865 M241.37931034482762 38.793103448275865 L250.00000000000003 38.793103448275865 M0 47.41379310344828 L8.620689655172415 47.41379310344828 M51.72413793103449 47.41379310344828 L60.344827586206904 47.41379310344828 M68.96551724137932 47.41379310344828 L77.58620689655173 47.41379310344828 M86.20689655172414 47.41379310344828 L163.79310344827587 47.41379310344828 M172.41379310344828 47.41379310344828 L181.0344827586207 47.41379310344828 M189.6551724137931 47.41379310344828 L198.27586206896552 47.41379310344828 M241.37931034482762 47.41379310344828 L250.00000000000003 47.41379310344828 M0 56.0344827586207 L60.344827586206904 56.0344827586207 M68.96551724137932 56.0344827586207 L77.58620689655173 56.0344827586207 M86.20689655172414 56.0344827586207 L94.82758620689656 56.0344827586207 M103.44827586206898 56.0344827586207 L112.0689655172414 56.0344827586207 M120.68965517241381 56.0344827586207 L129.31034482758622 56.0344827586207 M137.93103448275863 56.0344827586207 L146.55172413793105 56.0344827586207 M155.17241379310346 56.0344827586207 L163.79310344827587 56.0344827586207 M172.41379310344828 56.0344827586207 L181.0344827586207 56.0344827586207 M189.6551724137931 56.0344827586207 L250.00000000000003 56.0344827586207 M68.96551724137932 64.65517241379311 L86.20689655172414 64.65517241379311 M94.82758620689656 64.65517241379311 L120.68965517241381 64.65517241379311 M129.31034482758622 64.65517241379311 L146.55172413793105 64.65517241379311 M155.17241379310346 64.65517241379311 L163.79310344827587 64.65517241379311 M172.41379310344828 64.65517241379311 L181.0344827586207 64.65517241379311 M0 73.27586206896552 L8.620689655172415 73.27586206896552 M17.24137931034483 73.27586206896552 L60.344827586206904 73.27586206896552 M77.58620689655173 73.27586206896552 L112.0689655172414 73.27586206896552 M120.68965517241381 73.27586206896552 L137.93103448275863 73.27586206896552 M146.55172413793105 73.27586206896552 L155.17241379310346 73.27586206896552 M163.79310344827587 73.27586206896552 L172.41379310344828 73.27586206896552 M189.6551724137931 73.27586206896552 L232.7586206896552 73.27586206896552 M0 81.89655172413794 L25.862068965517246 81.89655172413794 M34.48275862068966 81.89655172413794 L43.10344827586207 81.89655172413794 M60.344827586206904 81.89655172413794 L68.96551724137932 81.89655172413794 M137.93103448275863 81.89655172413794 L172.41379310344828 81.89655172413794 M181.0344827586207 81.89655172413794 L224.1379310344828 81.89655172413794 M241.37931034482762 81.89655172413794 L250.00000000000003 81.89655172413794 M0 90.51724137931035 L8.620689655172415 90.51724137931035 M43.10344827586207 90.51724137931035 L60.344827586206904 90.51724137931035 M86.20689655172414 90.51724137931035 L94.82758620689656 90.51724137931035 M103.44827586206898 90.51724137931035 L112.0689655172414 90.51724137931035 M129.31034482758622 90.51724137931035 L146.55172413793105 90.51724137931035 M155.17241379310346 90.51724137931035 L163.79310344827587 90.51724137931035 M181.0344827586207 90.51724137931035 L189.6551724137931 90.51724137931035 M34.48275862068966 99.13793103448276 L43.10344827586207 99.13793103448276 M60.344827586206904 99.13793103448276 L68.96551724137932 99.13793103448276 M120.68965517241381 99.13793103448276 L146.55172413793105 99.13793103448276 M172.41379310344828 99.13793103448276 L181.0344827586207 99.13793103448276 M198.27586206896552 99.13793103448276 L215.51724137931038 99.13793103448276 M241.37931034482762 99.13793103448276 L250.00000000000003 99.13793103448276 M8.620689655172415 107.75862068965519 L25.862068965517246 107.75862068965519 M34.48275862068966 107.75862068965519 L60.344827586206904 107.75862068965519 M68.96551724137932 107.75862068965519 L86.20689655172414 107.75862068965519 M94.82758620689656 107.75862068965519 L112.0689655172414 107.75862068965519 M129.31034482758622 107.75862068965519 L155.17241379310346 107.75862068965519 M172.41379310344828 107.75862068965519 L181.0344827586207 107.75862068965519 M198.27586206896552 107.75862068965519 L206.89655172413796 107.75862068965519 M224.1379310344828 107.75862068965519 L241.37931034482762 107.75862068965519 M8.620689655172415 116.3793103448276 L17.24137931034483 116.3793103448276 M34.48275862068966 116.3793103448276 L51.72413793103449 116.3793103448276 M77.58620689655173 116.3793103448276 L86.20689655172414 116.3793103448276 M94.82758620689656 116.3793103448276 L129.31034482758622 116.3793103448276 M137.93103448275863 116.3793103448276 L146.55172413793105 116.3793103448276 M155.17241379310346 116.3793103448276 L232.7586206896552 116.3793103448276 M241.37931034482762 116.3793103448276 L250.00000000000003 116.3793103448276 M8.620689655172415 125.00000000000001 L34.48275862068966 125.00000000000001 M43.10344827586207 125.00000000000001 L68.96551724137932 125.00000000000001 M77.58620689655173 125.00000000000001 L103.44827586206898 125.00000000000001 M112.0689655172414 125.00000000000001 L137.93103448275863 125.00000000000001 M155.17241379310346 125.00000000000001 L163.79310344827587 125.00000000000001 M172.41379310344828 125.00000000000001 L181.0344827586207 125.00000000000001 M189.6551724137931 125.00000000000001 L224.1379310344828 125.00000000000001 M0 133.62068965517244 L8.620689655172415 133.62068965517244 M17.24137931034483 133.62068965517244 L25.862068965517246 133.62068965517244 M68.96551724137932 133.62068965517244 L77.58620689655173 133.62068965517244 M103.44827586206898 133.62068965517244 L120.68965517241381 133.62068965517244 M129.31034482758622 133.62068965517244 L146.55172413793105 133.62068965517244 M163.79310344827587 133.62068965517244 L172.41379310344828 133.62068965517244 M181.0344827586207 133.62068965517244 L189.6551724137931 133.62068965517244 M198.27586206896552 133.62068965517244 L206.89655172413796 133.62068965517244 M215.51724137931038 133.62068965517244 L224.1379310344828 133.62068965517244 M0 142.24137931034485 L8.620689655172415 142.24137931034485 M25.862068965517246 142.24137931034485 L43.10344827586207 142.24137931034485 M51.72413793103449 142.24137931034485 L60.344827586206904 142.24137931034485 M103.44827586206898 142.24137931034485 L112.0689655172414 142.24137931034485 M120.68965517241381 142.24137931034485 L129.31034482758622 142.24137931034485 M146.55172413793105 142.24137931034485 L155.17241379310346 142.24137931034485 M172.41379310344828 142.24137931034485 L181.0344827586207 142.24137931034485 M224.1379310344828 142.24137931034485 L232.7586206896552 142.24137931034485 M0 150.86206896551727 L17.24137931034483 150.86206896551727 M43.10344827586207 150.86206896551727 L51.72413793103449 150.86206896551727 M68.96551724137932 150.86206896551727 L86.20689655172414 150.86206896551727 M94.82758620689656 150.86206896551727 L112.0689655172414 150.86206896551727 M137.93103448275863 150.86206896551727 L146.55172413793105 150.86206896551727 M155.17241379310346 150.86206896551727 L181.0344827586207 150.86206896551727 M189.6551724137931 150.86206896551727 L198.27586206896552 150.86206896551727 M206.89655172413796 150.86206896551727 L224.1379310344828 150.86206896551727 M232.7586206896552 150.86206896551727 L250.00000000000003 150.86206896551727 M0 159.48275862068968 L8.620689655172415 159.48275862068968 M25.862068965517246 159.48275862068968 L43.10344827586207 159.48275862068968 M51.72413793103449 159.48275862068968 L60.344827586206904 159.48275862068968 M77.58620689655173 159.48275862068968 L86.20689655172414 159.48275862068968 M129.31034482758622 159.48275862068968 L163.79310344827587 159.48275862068968 M172.41379310344828 159.48275862068968 L181.0344827586207 159.48275862068968 M206.89655172413796 159.48275862068968 L215.51724137931038 159.48275862068968 M224.1379310344828 159.48275862068968 L232.7586206896552 159.48275862068968 M0 168.1034482758621 L8.620689655172415 168.1034482758621 M17.24137931034483 168.1034482758621 L43.10344827586207 168.1034482758621 M60.344827586206904 168.1034482758621 L86.20689655172414 168.1034482758621 M120.68965517241381 168.1034482758621 L137.93103448275863 168.1034482758621 M155.17241379310346 168.1034482758621 L163.79310344827587 168.1034482758621 M172.41379310344828 168.1034482758621 L181.0344827586207 168.1034482758621 M198.27586206896552 168.1034482758621 L224.1379310344828 168.1034482758621 M232.7586206896552 168.1034482758621 L241.37931034482762 168.1034482758621 M0 176.7241379310345 L8.620689655172415 176.7241379310345 M25.862068965517246 176.7241379310345 L43.10344827586207 176.7241379310345 M51.72413793103449 176.7241379310345 L60.344827586206904 176.7241379310345 M94.82758620689656 176.7241379310345 L103.44827586206898 176.7241379310345 M129.31034482758622 176.7241379310345 L137.93103448275863 176.7241379310345 M146.55172413793105 176.7241379310345 L155.17241379310346 176.7241379310345 M172.41379310344828 176.7241379310345 L215.51724137931038 176.7241379310345 M224.1379310344828 176.7241379310345 L232.7586206896552 176.7241379310345 M68.96551724137932 185.34482758620692 L86.20689655172414 185.34482758620692 M112.0689655172414 185.34482758620692 L129.31034482758622 185.34482758620692 M137.93103448275863 185.34482758620692 L163.79310344827587 185.34482758620692 M172.41379310344828 185.34482758620692 L181.0344827586207 185.34482758620692 M206.89655172413796 185.34482758620692 L215.51724137931038 185.34482758620692 M224.1379310344828 185.34482758620692 L232.7586206896552 185.34482758620692 M241.37931034482762 185.34482758620692 L250.00000000000003 185.34482758620692 M0 193.96551724137933 L60.344827586206904 193.96551724137933 M94.82758620689656 193.96551724137933 L137.93103448275863 193.96551724137933 M163.79310344827587 193.96551724137933 L181.0344827586207 193.96551724137933 M189.6551724137931 193.96551724137933 L198.27586206896552 193.96551724137933 M206.89655172413796 193.96551724137933 L224.1379310344828 193.96551724137933 M0 202.58620689655174 L8.620689655172415 202.58620689655174 M51.72413793103449 202.58620689655174 L60.344827586206904 202.58620689655174 M68.96551724137932 202.58620689655174 L86.20689655172414 202.58620689655174 M103.44827586206898 202.58620689655174 L120.68965517241381 202.58620689655174 M137.93103448275863 202.58620689655174 L146.55172413793105 202.58620689655174 M155.17241379310346 202.58620689655174 L181.0344827586207 202.58620689655174 M206.89655172413796 202.58620689655174 L224.1379310344828 202.58620689655174 M241.37931034482762 202.58620689655174 L250.00000000000003 202.58620689655174 M0 211.20689655172418 L8.620689655172415 211.20689655172418 M17.24137931034483 211.20689655172418 L43.10344827586207 211.20689655172418 M51.72413793103449 211.20689655172418 L60.344827586206904 211.20689655172418 M68.96551724137932 211.20689655172418 L94.82758620689656 211.20689655172418 M120.68965517241381 211.20689655172418 L155.17241379310346 211.20689655172418 M172.41379310344828 211.20689655172418 L241.37931034482762 211.20689655172418 M0 219.8275862068966 L8.620689655172415 219.8275862068966 M17.24137931034483 219.8275862068966 L43.10344827586207 219.8275862068966 M51.72413793103449 219.8275862068966 L60.344827586206904 219.8275862068966 M68.96551724137932 219.8275862068966 L94.82758620689656 219.8275862068966 M103.44827586206898 219.8275862068966 L112.0689655172414 219.8275862068966 M137.93103448275863 219.8275862068966 L146.55172413793105 219.8275862068966 M155.17241379310346 219.8275862068966 L163.79310344827587 219.8275862068966 M224.1379310344828 219.8275862068966 L250.00000000000003 219.8275862068966 M0 228.448275862069 L8.620689655172415 228.448275862069 M17.24137931034483 228.448275862069 L43.10344827586207 228.448275862069 M51.72413793103449 228.448275862069 L60.344827586206904 228.448275862069 M68.96551724137932 228.448275862069 L86.20689655172414 228.448275862069 M103.44827586206898 228.448275862069 L112.0689655172414 228.448275862069 M120.68965517241381 228.448275862069 L146.55172413793105 228.448275862069 M163.79310344827587 228.448275862069 L172.41379310344828 228.448275862069 M181.0344827586207 228.448275862069 L215.51724137931038 228.448275862069 M232.7586206896552 228.448275862069 L241.37931034482762 228.448275862069 M0 237.06896551724142 L8.620689655172415 237.06896551724142 M51.72413793103449 237.06896551724142 L60.344827586206904 237.06896551724142 M77.58620689655173 237.06896551724142 L86.20689655172414 237.06896551724142 M94.82758620689656 237.06896551724142 L112.0689655172414 237.06896551724142 M120.68965517241381 237.06896551724142 L146.55172413793105 237.06896551724142 M163.79310344827587 237.06896551724142 L181.0344827586207 237.06896551724142 M198.27586206896552 237.06896551724142 L224.1379310344828 237.06896551724142 M232.7586206896552 237.06896551724142 L241.37931034482762 237.06896551724142 M0 245.68965517241384 L60.344827586206904 245.68965517241384 M68.96551724137932 245.68965517241384 L77.58620689655173 245.68965517241384 M86.20689655172414 245.68965517241384 L103.44827586206898 245.68965517241384 M112.0689655172414 245.68965517241384 L129.31034482758622 245.68965517241384 M146.55172413793105 245.68965517241384 L172.41379310344828 245.68965517241384 M198.27586206896552 245.68965517241384 L215.51724137931038 245.68965517241384 M224.1379310344828 245.68965517241384 L232.7586206896552 245.68965517241384 "
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
              ]
            }
            stroke={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            strokeLinecap={0}
            strokeWidth="8.620689655172415"
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      {
        "alignItems": "flex-end",
        "flex": 1,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 2,
          },
          {
            "flex": 1,
            "marginHorizontal": 5,
            "marginVertical": 5,
          },
          false,
        ]
      }
      testID="RNE_BUTTON_WRAPPER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": false,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 5,
              "borderWidth": 0,
              "flexDirection": "row-reverse",
              "justifyContent": "center",
              "padding": 8,
              "paddingHorizontal": 10,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "white",
                  "fontSize": 50,
                },
                undefined,
                {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 2,
          },
          {
            "flex": 1,
            "marginHorizontal": 5,
            "marginVertical": 5,
          },
          false,
        ]
      }
      testID="RNE_BUTTON_WRAPPER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": false,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 5,
              "borderWidth": 0,
              "flexDirection": "row-reverse",
              "justifyContent": "center",
              "padding": 8,
              "paddingHorizontal": 10,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "white",
                  "fontSize": 50,
                },
                undefined,
                {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
